packages/modernjs/src/cli/index.ts

```diff
@@ -0,0 +1,253 @@
+  patchWebpackConfig,
+  getIPV4,
+} from './utils';
+import { updateStatsAndManifest } from './manifest';
+import { MODERN_JS_SERVER_DIR, PLUGIN_IDENTIFIER } from '../constant';
+
+const SSR_PLUGIN_IDENTIFIER = 'mfPluginSSR';
+const isDev = process.env.NODE_ENV === 'development';
+
+export const moduleFederationPlugin = (
+  userConfig: PluginOptions = {},
+): CliPlugin<AppTools> => ({
+  name: '@modern-js/plugin-module-federation',
+  setup: async ({ useConfigContext, useAppContext }) => {
+    const modernjsConfig = useConfigContext();
+    const enableSSR = Boolean(modernjsConfig?.server?.ssr);
+    const mfConfig = await getMFConfig(userConfig);
+    let outputDir = '';
+
+    let browserPlugin: BundlerPlugin;
+    let nodePlugin: BundlerPlugin;
+    return {
+      config: async () => {
+        const bundlerType =
+          useAppContext().bundlerType === 'rspack' ? 'rspack' : 'webpack';
+
+        const WebpackPluginConstructor =
+          userConfig.webpackPluginImplementation ||
+          WebpackModuleFederationPlugin;
+        const RspackPluginConstructor =
+          userConfig.rspackPluginImplementation || RspackModuleFederationPlugin;
+
+        const MFBundlerPlugin =
+          bundlerType === 'rspack'
+            ? RspackPluginConstructor
+            : WebpackPluginConstructor;
+
+        if (enableSSR) {
+          process.env['MF_DISABLE_EMIT_STATS'] = 'true';
+          process.env['MF_SSR_PRJ'] = 'true';
+        }
+
+        const modifyBundlerConfig = <T extends 'webpack' | 'rspack'>(
+          config: ConfigType<T>,
+          isServer: boolean,
+        ) => {
+          const envConfig = getTargetEnvConfig(mfConfig, isServer);
+          if (isServer) {
+            nodePlugin = new MFBundlerPlugin(envConfig);
+            // @ts-expect-error the compiler version can not be equal, so it usually throw type errors
+            config.plugins?.push(nodePlugin);
+            // @ts-expect-error the compiler version can not be equal, so it usually throw type errors
+            config.plugins?.push(new StreamingTargetPlugin(envConfig));
+            if (isDev) {
+              // @ts-expect-error the compiler version can not be equal, so it usually throw type errors
+              config.plugins?.push(new EntryChunkTrackerPlugin());
+            }
+          } else {
+            outputDir =
+              config.output?.path || path.resolve(process.cwd(), 'dist');
+            browserPlugin = new MFBundlerPlugin(envConfig);
+            // @ts-expect-error the compiler version can not be equal, so it usually throw type errors
+            config.plugins?.push(browserPlugin);
+          }
+
+          patchWebpackConfig({
+            bundlerConfig: config,
+            isServer,
+            modernjsConfig,
+            mfConfig: envConfig,
+          });
+        };
+
+        const ipv4 = getIPV4();
+
+        return {
+          tools: {
+            rspack(config) {
+              if (enableSSR) {
```

Comment:
Support for rspack versus webpack what the difference is

Reply:
Is it mainly the following async logic?

Reply:
Yes,  rspack not implement the AsyncBoundaryPlugin yet , so i need to do some work for this